src/groups/group.jl

```diff
@@ -1121,6 +1186,43 @@ end
 direction_and_side(::GroupExponentialRetraction{D}) where {D} = D()
 direction_and_side(::GroupLogarithmicInverseRetraction{D}) where {D} = D()
 
+function log(::TraitList{<:IsGroupManifold}, G::AbstractDecoratorManifold, p, q)
+    # TODO: use the following version when `allocate_result` is fixed
```

Comment:
Maybe more something for Mateusz – what needs to be fixed here?

Reply:
One problem is that `allocate_result` does not respect the underlying numerical type of the group. So if `G` is a complex group, the allocated variable is real. There are other problems: some groups do not use vectors (e.g. `CircleGroup`, and allocation does not work.

Reply:
Mateus is the allocation expert (it works for the circle and complex manifolds for example), so then we should adapt that for sure.

Reply:
I will take a look what is the issue with the `allocate_result` variant. We generally assume in generic code that points and tangent vectors are mutable and make manifold-specific adjustments when they are not (like for example for `CircleGroup`).

Reply:
Here is an example:
```julia
Q2 = Unitary(2, ℍ)
allocate_result(Q2, log) # matrix of floats instead of quaternions
```
I suggest to remove this comment for now.

Reply:
I've removed the TODO. This can be fixed later.

Reply:
@Mateusz should we open an issue for this to keep that in mind? Besides that this comment is finished :)

Reply:
I will fix that quickly in https://github.com/JuliaManifolds/ManifoldsBase.jl/pull/192 since that is also related to allocation and not merged yet.

Reply:
Then this point here is resolved.

Reply:
Yes, I've just pushed the fix.